### DISCUSSION TITLE
How to include symlinks in a python wheel?

### DISCUSSION THREAD

**Discussion by dranobob** [python-poetry/poetry#3564] [Q&A]
I have a python package that I want to include some specific files from a submodule without including the entire package, so the structure looks like:

```
foo_project
 |
 +-- submodule_files
     +-- bar.json
     +-- other_stuff
 +-- foo
     +-- __init__.py
     +-- foo.py
     +-- <symlink to bar.json>
```

The poetry toml file has

```
packages = [
    { include = "foo" }
]
```

When the wheel is created using poetry build, the linked files are copied, but not the links themselves

```
 foo_wheel
 |
 +-- submodule_files
     +-- bar.json
 +-- foo
     +-- __init__.py
     +-- foo.py
```
(left out info directory in wheel for brevity)

Therefore once installed, the python package is missing the extra files

**Comment by jaghaimo** [comment]
Struggling with this as well - did you even find out how? So far, the best I could do was a post-install script that recreates the missing symlinks.

**Comment by neersighted** [comment]
This is not possible [per the wheel standard](https://discuss.python.org/t/symbolic-links-in-wheels/1945).

**Comment by arcivanov** [comment]
It's possible with wheel-axle, a drop-in replacement for wheels: https://github.com/karellen/wheel-axle
